syft/workers/base.py

```diff
@@ -951,7 +951,7 @@ def search(self, query: Union[List[Union[str, int]], str, int]) -> List[PointerT
                 # set garbage_collect_data to False because if we're searching
                 # for a tensor we don't own, then it's probably someone else's
                 # decision to decide when to delete the tensor.
-                ptr = obj.create_pointer(garbage_collect_data=False, owner=sy.local_worker).wrap()
+                ptr = obj.create_pointer(garbage_collect_data=False, owner=sy.local_worker, location=self).wrap()
```

Comment:
Why did you need to change this `location` argument?

Reply:
because if the location wasn't passed it would assume the location is the local worker, which isn't correct. the search result would be always me | id -> me | id_at_location